qiskit_machine_learning/kernels/algorithms/quantum_kernel_trainer.py

```diff
@@ -0,0 +1,225 @@
+            feature_map=...,
+            user_parameters=...,
+            quantum_instance=...
+        )
+
+        loss_func = ...
+        optimizer = ...
+        initial_point = ...
+
+        qk_trainer = QuantumKernelTrainer(
+                                        quantum_kernel=quant_kernel,
+                                        loss=loss_func,
+                                        optimizer=optimizer,
+                                        initial_point=initial_point,
+                                        )
+        qkt_results = qk_trainer.fit_kernel(X_train, y_train)
+        optimized_kernel = qkt_results.quantum_kernel
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: QuantumKernel,
+        loss: Union[str, Callable[[Sequence[float]], float]] = "svc_loss",
+        optimizer: Optimizer = SPSA(),
+        initial_point: Optional[Sequence[float]] = None,
+    ):
+        """
+        Args:
+            quantum_kernel: QuantumKernel to be trained
+            loss (Callable[[Sequence[float]], float] or str):
+                str: Loss functions available via string: {'svc_loss: SVCLoss()).
+                    If a string is passed as the loss function, then the underlying
+                    KernelLoss object will exhibit default behavior.
+                Callable[[Sequence[float]], float]: A callable loss function which takes
+                    a vector of parameter values as input and returns a loss value (float)
+            optimizer: An instance of ``Optimizer`` to be used in training. Since no
+                analytical gradient is defined for kernel loss functions, gradient-based
+                optimizers are not recommended for training kernels.
+            initial_point: Initial point from which the optimizer will begin.
+
+        Raises:
+            ValueError: unknown loss function
+        """
+        # Class fields
+        self._quantum_kernel = quantum_kernel
+        self._optimizer = optimizer
+        self._initial_point = initial_point
+        self._loss = None
+
+        # Setters
+        self.loss = loss
+
+    @property
+    def quantum_kernel(self) -> "QuantumKernel":
+        """Return the quantum kernel object."""
+        return self._quantum_kernel
+
+    @quantum_kernel.setter
+    def quantum_kernel(self, quantum_kernel: "QuantumKernel") -> None:
+        """Set the quantum kernel."""
+        self._quantum_kernel = quantum_kernel
+
+    @property
+    def loss(self) -> Union[str, Callable[[Sequence[float]], float]]:
+        """Return the loss object."""
+        return self._loss
+
+    @loss.setter
+    def loss(self, loss: Union[str, Callable[[Sequence[float]], float]]) -> None:
+        """Set the loss."""
+        if isinstance(loss, str):
+            self._loss = loss.lower()
+            if self._loss == "svc_loss":
+                pass
+            else:
+                raise ValueError(f"Unknown loss {loss}!")
```

Comment:
I'm not sure I follow your if/else comment. This logic looks fine to me, what am I missing?

Reply:
I mean `if` clause has just `pass`, while `else` throws an error. Maybe better to swap them and make `if` to throw an error and then there's no need for `else`.

Reply:
Oh yes, I set it up in this funky way to lay the way for future losses. Since there is only one right now, the logic is a bit strange.

Reply:
In the future, with more than one value you can do something like an if not self._loss in ["a", "b", "c"] etc so as to throw an exception if its not a recognised one, assuming its all done the same way. So that reversal and throwing an error if its not in the recognized set is still a simple one line thing and does not require a complicated if/else structure. Assuming this is all this if test is expected to do based on type.